### PR TITLE
[DOC beta] Update Route#loading example

### LINKED ISSUE #13341: [DOC] Need better example for Route#loading
https://github.com/emberjs/ember.js/blob/v2.5.0/packages/ember-routing/lib/system/route.js#L615-L622
Uses deprecated Ember.View API in the example. I would bet this is confusing to new users coming to Ember and seeing this hook.

### PR DESCRIPTION
**Pull request by duggiefresh**
Fixes #13341

### CODE REVIEW

File: packages/ember-routing/lib/system/route.js [hunk 1]
```diff
@@ -605,23 +605,19 @@ var Route = EmberObject.extend(ActionHandler, Evented, {
...
-          var view = Ember.View.create({
-            classNames: ['app-loading']
-          })
-          .append();
+          let controller = this.controllerFor('foo');
+          controller.set('currentlyLoading', true);
 
-          this.router.one('didTransition', function() {
-            view.destroy();
+          transition.promise.finally(function() {
```
Comment (stefanpenner) [inline packages/ember-routing/lib/system/route.js]: is the promise public?

Comment (duggiefresh) [inline packages/ember-routing/lib/system/route.js]: [Looking at the router, it is marked as `public`](https://github.com/tildeio/router.js/blob/master/lib/router/transition.js#L104-L115) :smile:

Comment (Serabe) [inline packages/ember-routing/lib/system/route.js]: It is, but as DOC says, it is there if you want to pass around a promise without passing the full transition. In this case,  I think [`transition.finally`](https://github.com/tildeio/router.js/blob/master/lib/router/transition.js#L167-L182) would do too.

Comment (duggiefresh) [inline packages/ember-routing/lib/system/route.js]: Good call, will update :+1:

Comment (Serabe) [inline packages/ember-routing/lib/system/route.js]: Thank you.

**Comment by rwjblue** [comment]
Thanks @duggiefresh!

**Pull request merged**